projects/archiver/media.ts

```diff
@@ -0,0 +1,45 @@
+import { Front, ImageElement, CAPIArticle, Image } from './common'
+import { unnest } from 'ramda'
+import { getImage, getColours } from './downloader'
+import { hasFailed, attempt } from '../backend/utils/try'
+import { upload } from './upload'
+export const getImagesFromArticle = (article: CAPIArticle): Image[] => {
+    const image = article.image ? [article.image] : []
+    const elements = article.type === 'article' ? article.elements : []
+    const images = elements
+        .filter((element): element is ImageElement => element.id === 'image')
+        .map(element => element.src)
+
+    return [...images, ...image]
+}
+
+export const getImagesFromFront = (front: Front): Image[] => {
+    const allCards = unnest(front.collections.map(_ => _.cards))
+    const articles = unnest(allCards.map(_ => Object.values(_.articles)))
+    return unnest(articles.map(getImagesFromArticle))
+}
+
+export const uploadImage = async (id: string, image: Image) => {
+    const allSizes = await getImage(id, image)
+
+    const [colourPath, colours] = await getColours(id, image)
```

Comment:
For the uninitiated, is this a cached JSON of colours for an image in S3 somewhere?